### PR TITLE
feat: streaming updates python client [MD-246]

### PR DESCRIPTION
**Pull request by rb-determined-ai**
This includes a new code generator.

Our existing code generation is proto -> openapi -> tweaks -> python, with an alternate proto -> go path for the server code.

This is more direct, it's go -> python; the go code is the source of truth and we can write the generator in go to use go's ast package.

## Test Plan

Feature is not yet user-facing.  Automated testing is sufficient.

## Commentary

I configured the PR against the streaming updates "core functionality" feature branch lands for reviewability, but I plan on landing it to main after that branch lands.

### CODE REVIEW

**Review by gt2345 — APPROVED** [review]
Works as expected

**Review by azhou-determined — APPROVED** [review]

File: master/cmd/stream-gen/main.go [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+	content, err := genPython(results)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+
+	// write to output
+	if output == "" {
+		// write to stdout
+		_, err := os.Stderr.Write(content)
```
Comment (jgongd) [inline master/cmd/stream-gen/main.go:479]: Did you intend to use `os.Stdout`?

Comment (rb-determined-ai) [reply master/cmd/stream-gen/main.go:486]: yes, good catch

File: master/cmd/stream-gen/main.go [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+All structs in the input files IN.GO... which contain special 'determined:stream-gen' comments will
+be included in the generated output.
+
+Presently the only output language is --python.
+
+Output will be written to stdout, or a location specified by --output.  The OUTPUT will only be
+overwritten if it would be modified.
+
+If --stamp is provided, the STAMP file will be touched after a successful run, which is useful for
+integration with build systems.
```
Comment (jgongd) [inline master/cmd/stream-gen/main.go:411]: I'm curious how the stamp file is used for integration with build systems. Is there an example in the code base that I can take a look at?

**Review by jgongd — APPROVED** [review]

File: master/cmd/stream-gen/main.go (hunk 2 above)
Comment (rb-determined-ai) [reply master/cmd/stream-gen/main.go]: you know, looking at this... there isn't a good reason for a `--stamp` output with how this function behaves.  I'll just remove this.

**Comment by rb-determined-ai** [comment]
I need to figure out the CI situation here before landing this.

The code generation may complicate the relationships between stages.  Ugh.

**Comment by rb-determined-ai** [comment]
> I need to figure out the CI situation here before landing this.
> 
> The code generation may complicate the relationships between stages. Ugh.

no... that's not right.

CI is fine.  Since we still commit generated code, the `make -C master check` step will additional check that the stream-gen --python output is correct.  Nobody else has a reason to regenerate that file.

I will include `make -C master stream-gen` as part of the devcluster `p` keybinding though.

**Pull request merged**